superset-frontend/src/SqlLab/components/SaveDatasetModal/index.tsx

```diff
@@ -148,10 +172,10 @@ export const SaveDatasetModal: FunctionComponent<SaveDatasetModalProps> = ({
   const handleOverwriteDataset = async () => {
     const [, key] = await Promise.all([
       updateDataset(
-        query.dbId,
+        datasource.dbId,
         datasetToOverwrite.datasetid,
-        query.sql,
-        query.results.selected_columns.map(
+        datasource.sql,
+        datasource?.columns?.map(
```

Comment:
we have some datasources that seem optional here (via the optional chaining) and some that aren't. Is that intentional?